### PR TITLE
First pass at some more detailed documentation

### PR DESCRIPTION
**Pull request by shaladdle**
A start on #57

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -131,11 +131,67 @@ fn main() {
...
+use tarpc::util::Message;
+
+service! {
+    rpc hello(name: String) -> String | Message
+}
+```
+
+`tarpc::util::Message` is just a wrapper around string that implements `std::error::Error` provided
+for service implementations that don't require complex error handling. The pipe is used as syntax
+for specifying either both the result type and the error type in a way that's agnostic of whether
```
Comment (tikue) [inline README.md]: Typo: either

File: README.md [hunk 2]
```diff
@@ -131,11 +131,67 @@ fn main() {
...
+the service implementation is synchronous or uses futures. Note that in the simpler examples in the
+readme, no pipe is used, and the macro automatically chooses `tarpc::util::Never` as the error type.
+
+The above declaration declaration would produce the following synchronous service trait:
```
Comment (tikue) [inline README.md]: Typo

**Comment by shaladdle** [comment]
I addressed your comments I think. Is there some button I'm supposed to click to tell you to look again?

**Comment by shaladdle** [comment]
Tim advised me offline to merge this.

**Pull request merged**